lib/vmpooler/providers/base.rb

```diff
@@ -11,12 +11,16 @@ class Base
         attr_reader :metrics
         # Provider options passed in during initialization
         attr_reader :provider_options
+        # Hash for tracking hosts for deployment
```

Comment:
These settings seem vsphere specific.  Perhaps instead of modifying base, which affects all providers, move them there